presentation/src/main/java/com/architect/coders/mu8/characters/detail/CharactersDetailActivity.kt

```diff
@@ -1,14 +1,55 @@
 package com.architect.coders.mu8.characters.detail
 
-import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import androidx.databinding.DataBindingUtil
+import androidx.lifecycle.Observer
+import androidx.recyclerview.widget.LinearLayoutManager
 import com.architect.coders.mu8.R
+import com.architect.coders.mu8.data.DataApp
+import com.architect.coders.mu8.data.characters.CharactersRepositoryImpl
+import com.architect.coders.mu8.data.charactersDetail.CharacterDetailRepositoryImpl
+import com.architect.coders.mu8.databinding.ActivityCharactersDetailBinding
+import com.architect.coders.mu8.utils.getViewModel
+import com.architect.codes.mu8.characters.CharactersUseCaseImpl
+import com.architect.codes.mu8.charactersDetail.CharacterDetalUseCaseImpl
+import com.architect.codes.mu8.comics.Comic
 
 class CharactersDetailActivity : AppCompatActivity() {
 
+    private lateinit var viewModel: CharactersDetailViewModel
+    private lateinit var binding: ActivityCharactersDetailBinding
+    private lateinit var adapter: CharacterDetailAdapter
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_characters_detail)
+        binding = DataBindingUtil.setContentView(this, R.layout.activity_characters_detail)
+
+        initVM(intent.getLongExtra(CHARACTER, -1))
```

Comment:
Please be specific with the method names, what is VM?

Reply:
vm = viewmodel, change by initViewModel